MorphicServer/Preferences.cs

```diff
@@ -33,15 +32,15 @@ namespace MorphicServer
     public class Preferences : Record
     {
         [JsonPropertyName("user_id")]
-        public string? UserId { get; set; }
+        public string UserId { get; set; } = null!;
 
         /// <summary>The user's default preferences</summary>
         // Stored as a serialized JSON string in the mongo database because keys might contain dots,
         // and mongoDB doesn't allow dots in field keys.  Since we're unlikely to need to run queries
         // within the solution preferences, we don't lose any functionality by storing serialized JSON.
         [BsonSerializer(typeof(Database.JsonSerializer<Dictionary<string, SolutionPreferences>>))]
         [JsonPropertyName("default")]
-        public Dictionary<string, SolutionPreferences>? Default { get; set; }
+        public Dictionary<string, SolutionPreferences> Default { get; set; } = null!;
```

Comment:
If we're doing a `PreferencesRequest`, then this change here doesn't have any effect for enforcing nonnull.  I'd prefer to either keep it nullable or not initialize to `null`.  Thoughts?

Reply:
Makes sense. Reverted.